controllers/texttemplate_controller.go

```diff
@@ -189,37 +206,22 @@ func (r *TextTemplateReconciler) doReconcile(ctx context.Context, tt *templatesv
 	return nil
 }
 
+func (r *TextTemplateReconciler) finalize(ctx context.Context, obj *templatesv1alpha1.TextTemplate) (ctrl.Result, error) {
+	r.watchesUtil.removeWatchesForTemplate(client.ObjectKeyFromObject(obj))
+
+	// Remove our finalizer from the list and update it
+	controllerutil.RemoveFinalizer(obj, templatesv1alpha1.TextTemplateFinalizer)
+	if err := r.Update(ctx, obj, client.FieldOwner(r.FieldManager)); err != nil {
```

Comment:
Is it possible to remove the finalizer with a patch? You add it with a patch, and I am unsure it you can remove it with an update. You should also check that the finalizer is removed when the object is "deleted" in tests. It is crucial that finalizers work! I try to avoid finalizers, but in this case I think they are required - at least for now.

Reply:
I switched to using patches now and added some simple testing for final removal, which implies proper removal of finalizers. I'll add more tests in regard to pruning (where finalizers are even more important) in a future PR.